File: CalSnap/Core/Models/MealEntry.swift
import Foundation
import SwiftData

@Model
final class MealEntry {
    @Attribute(.unique) var id: UUID
    var userId: UUID
    var timestamp: Date
    var mealType: MealType
    @Attribute(.externalStorage) var photoData: Data?
    var textDescription: String?
    var totalCalories: Int
    var totalProteinG: Double
    var totalCarbsG: Double
    var totalFatG: Double
    var totalFiberG: Double
    var geminiConfidence: Double
    var isManuallyAdjusted: Bool
    var estimationNotes: String?
    @Relationship(deleteRule: .cascade) var items: [FoodItem]

    init(
        id: UUID = UUID(),
        userId: UUID,
        timestamp: Date = Date(),
        mealType: MealType = .suggested(for: Date()),
        photoData: Data? = nil,
        textDescription: String? = nil,
        totalCalories: Int = 0,
        totalProteinG: Double = 0,
        totalCarbsG: Double = 0,
        totalFatG: Double = 0,
        totalFiberG: Double = 0,
        geminiConfidence: Double = 0,
        isManuallyAdjusted: Bool = false,
        estimationNotes: String? = nil,
        items: [FoodItem] = []
    ) {
        self.id = id
        self.userId = userId
        self.timestamp = timestamp
        self.mealType = mealType
        self.photoData = photoData
        self.textDescription = textDescription
        self.totalCalories = totalCalories
        self.totalProteinG = totalProteinG
        self.totalCarbsG = totalCarbsG
        self.totalFatG = totalFatG
        self.totalFiberG = totalFiberG
        self.geminiConfidence = geminiConfidence
        self.isManuallyAdjusted = isManuallyAdjusted
        self.estimationNotes = estimationNotes
        self.items = items
    }
}

File: calsnap-web/lib/models/meal-entry.ts
import type { MealType } from '@/lib/models/meal-type';
import type { FoodItem } from '@/lib/models/food-item';

/** Logged meal entry. Totals and item weights are in metric units. */
export interface MealEntry {
  id: string;
  userId: string;
  timestamp: Date;
  mealType: MealType;
  /** Firebase Storage path; populated when a photo is uploaded (W04). */
  photoStoragePath?: string;
  textDescription?: string;
  totalCalories: number;
  totalProteinG: number;
  totalCarbsG: number;
  totalFatG: number;
  totalFiberG: number;
  geminiConfidence: number;
  isManuallyAdjusted: boolean;
  estimationNotes?: string;
  items: FoodItem[];
}
